### PR TITLE
Dev: Implement `aimag` directly in ASR

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Fixes https://github.com/lfortran/lfortran/issues/3373.

See: https://github.com/lfortran/lfortran/pull/3361#discussion_r1482504613

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
Ready!

**Review by certik — COMMENTED** [review]
Looks good so far.

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 1]
```diff
@@ -2283,15 +2283,10 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
-                args.push_back(al, val);
-                ASR::asr_t* create_fn = create_func(al, loc, args, diag);
-                if (create_fn == nullptr) {
-                    throw SemanticAbort();
-                }
-                ASR::expr_t *im = ASRUtils::EXPR(create_fn);
+                ASR::ttype_t *real_type = ASRUtils::TYPE(ASR::make_Real_t(al, loc,
+                    ASRUtils::extract_kind_from_ttype_t(ASRUtils::expr_type(val))));
+                ASR::expr_t *im = ASRUtils::EXPR(ASR::make_ComplexIm_t(al, loc,
+                    val, real_type, nullptr));
```
[the remark this answers is not in the document]
Comment (certik) [reply src/lfortran/semantics/ast_body_visitor.cpp:2289]: It's weird that no test failed, so we probably also need to add a test.

[the remark this answers is not in the document]
Comment (Thirumalai-Shaktivel) [reply src/lfortran/semantics/ast_body_visitor.cpp:2289]: Sorry, I didn't get it. Why would tests fail?
We just replaced the `aimag` usage with `ComplexIm`, so we should expect no test to fail, right?

[the remark this answers is not in the document]
Comment (certik) [reply src/lfortran/semantics/ast_body_visitor.cpp:2289]: My bad, I thought the code above was calling into the C runtime library for the functions that you removed. But they are instead calling `ComplexIm`, so all is good.

@Shaikh-Ubaid what needs to be fixed above?

**Comment by certik** [comment]
Can you please resolve the conflicts?

**Comment by Thirumalai-Shaktivel** [comment]
Ready!

**Review by certik — APPROVED** [review]

**Pull request merged**